temporian/core/compilation.py

```diff
@@ -24,57 +24,103 @@
 # TODO: unify the fn's output type with run's EvaluationQuery, and add it to the
 # public API so it shows in the docs.
 # TODO: make compile change the fn's annotations to EventSetOrNode
-def compile(fn: T) -> T:
+def compile(
+    fn: Optional[Callable] = None, *, verbose: Optional[int] = None
+) -> Any:
     """Compiles a Temporian function.
 
-    A Temporian function is a function that takes EventSetNodes as arguments and
-    returns EventSetNodes as outputs. Compiling it enables it to perform eager
-    evaluation, i.e., receive and return EventSets instead of EventSetNodes.
+    A Temporian function is a function that takes
+    [`EventSetNodes`][temporian.EventSetNode] as arguments and returns
+    [`EventSetNodes`][temporian.EventSetNode] as outputs. Compiling it enables
+    it to perform eager evaluation, i.e., receive and return
```

Comment:
The other side of the coin is that with @compile, you run a function operating on tp.EventSet faster, since the operators run both on Node and EventSet. I wonder if this is also worth mentionning.